tests/Fixtures/Stories/StoryWhichReadsItsOwnPool.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Zenstruck\Foundry\Tests\Fixtures\Stories;
+
+use Zenstruck\Foundry\Story;
+use Zenstruck\Foundry\Tests\Fixtures\Factories\CategoryFactory;
+
+final class StoryWhichReadsItsOwnPool extends Story
+{
+    public function build(): void
+    {
+        $this->addState('php', CategoryFactory::new()->create(['name' => 'php']), 'some-pool');
+        $this->addState('symfony', CategoryFactory::new()->create(['name' => 'symfony']), 'some-pool');
+
+        // story can access its own state
+        $this->addState('php2', self::get('php'), 'some-pool');
```

Comment:
```suggestion
        $this->addState('php2', self::getState('php'), 'some-pool');
```

Should fix the self-deprecation.

Reply:
fixed